### ISSUE TITLE
Xcode 11 support

### ISSUE THREAD

**Issue by NiekBogaard** [tadija/bitrise-step-xcode-project-info#2]
This library does not works any more on xcode 11 >. Due to xcode changes.
See: https://stackoverflow.com/questions/56722677/how-to-read-current-app-version-in-xcode-11-with-script

**Comment by tadija** [comment]
Hi @NiekBogaard,

I checked this and it is not true. Xcode 11 is still using `CFBundleVersion` and `CFBundleShortVersionString` in **Info.plist** file of the project. **Versioning** section in Build Settings is nothing new, it already exists since much older versions of Xcode. Check out [this gist](https://gist.github.com/tadija/7247e57e89187c0e9c1d) if you want to find out more about how to use "Apple Generic" versioning system with **agvtool**.

**Comment by Mayzie** [comment]
@tadija It is true. While it is still using `CFBundleShortVersionString`, any changes to the version number within XCode 11 will set it to `${MARKETING_VERSION}`, and this is what this step will now retrieve and breaks backward compatibility.

Our usage is to retrieve the version string from our main scheme, to then set that version in an app extension (notably a [notification service extension](https://documentation.onesignal.com/docs/ios-sdk-setup#section-1-add-notification-service-extension)). As it is, it is currently just setting it to `${MARKETING_VERSION}`, which is breaking our ability to automatically push our app to TestFlight as the versions are now not unified. This step was working fine prior to XCode 11, and now it is breaking with XCode 11.

Is there any documentation to suggest what we should instead do to support our use case, or what we should switch to instead?

**Comment by valerio-bettini** [comment]
Any update on this? I can confirm the step is pulling $MARKETING_VERSION, which is not correct!

**Comment by iMac0de** [comment]
This comment of the following thread retrieve the marketing version:
https://stackoverflow.com/a/58769925/4271442

**Comment by lucaskuzma** [comment]
FWIW, similar problem resolved here: https://github.com/denys-meloshyn/bitrise-step-git-tag-project-version-and-build-number/issues/12#issuecomment-560707235

**Comment by kkizlaitis** [comment]
In our case, we use a lot of targets, which all have `$(inherited)` values. When updating the app, we only need to change the _Marketing Version_ for the project itself. To extract the `MARKETING_VERSION` that is not `$(inherited)` we use this bash script:
```bash
MARKETING_VERSION=""

marketing_version_regex="MARKETING_VERSION = \"([^$]*)\";" # $(inherited) is not a valid value
project_file=$(<${PROJECT_PATH}/project.pbxproj)

if [[ $project_file =~ $marketing_version_regex ]]; then
	MARKETING_VERSION="${BASH_REMATCH[1]}";
else
	MARKETING_VERSION="unknown";
fi

echo $MARKETING_VERSION;
```

**Comment by tadija** [comment]
This should finally be fixed by merging PR #3 from @dlajarretie, thanks!
